ambry-cloud/src/main/java/com/github/ambry/cloud/AzureStorageReplicationManager.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2023 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.cloud;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.DataNodeId;
+import com.github.ambry.clustermap.ReplicaSyncUpManager;
+import com.github.ambry.clustermap.VcrClusterParticipant;
+import com.github.ambry.commons.ResponseHandler;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.ReplicationConfig;
+import com.github.ambry.config.ServerConfig;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.network.NetworkClient;
+import com.github.ambry.network.NetworkClientFactory;
+import com.github.ambry.notification.NotificationSystem;
+import com.github.ambry.replication.FindTokenHelper;
+import com.github.ambry.replication.ReplicaThread;
+import com.github.ambry.replication.ReplicationException;
+import com.github.ambry.replication.ReplicationManager;
+import com.github.ambry.replication.ReplicationMetrics;
+import com.github.ambry.server.StoreManager;
+import com.github.ambry.store.MessageInfo;
+import com.github.ambry.store.StoreKeyConverter;
+import com.github.ambry.store.StoreKeyConverterFactory;
+import com.github.ambry.store.StoreKeyFactory;
+import com.github.ambry.store.Transformer;
+import com.github.ambry.utils.Time;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Predicate;
+
+
+/**
+ * Manages replication from servers to VCR
+ */
+public class AzureStorageReplicationManager extends VcrReplicationManager {
```

Comment:
Any new method we need for the AzureStorageReplicatonManager, other than the methods from the parent class?

Same question for AzureStorageReplicationThread. Although we have V1 and V2, if the functionality is the same for some of the high level classes, we don't need create the duplicated classes.

Reply:
Yes and yes.

AzureStorageReplicationManager over writes getReplicaThread to return AzureStorageReplicationThread. Going forward, it will schedule a new type of thread for compaction.

AzureStorageReplicationThread over writes advanceToken and simplifies token persistence. It brings it to the foreground and makes debugging a lot more easier. Otherwise, the token persistence thread runs somewhere in the background and doesn't persist tokens as expected. I expect the token to be persisted after each batch of replication messages deterministically and would like to see it advance in the web-portal during a live debugging session.